From 838e8d873223d5c514eece5ad5fb11806dbaae77 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 3 Dec 2018 16:47:33 +0200
Subject: [PATCH 0823/1051] net: mvpp2: abort mac-config if already in up-state

Mac-config is called on UP-request by phylink state machine.
Check and do NOT execute the configuration if link if already UP.

This is needed for proper stop_dev/start_dev in mtu and ringsize change
on interfaces with phylink-polling.

Change-Id: Id5cd58b6cc1b3fddeb15458495d0edb70c44c081
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61478
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 81ea93e22ba5..5c3a314deb20 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -5611,6 +5611,10 @@ static void mvpp2_mac_config(struct net_device *dev, unsigned int mode,
 		return;
 	}
 
+	/* Mac-config is called on UP-request. Don't repeat if already UP */
+	if (state->link && netif_carrier_ok(dev) && port->has_phy)
+		return;
+
 	mvpp2_tx_stop_all_queues(port->dev);
 
 	/* Make sure the port is disabled when reconfiguring the mode */
-- 
2.17.1

